src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -197,10 +216,6 @@ internal void Append(JsonTokenType tokenType, int startLocation, int length)
                     (tokenType == JsonTokenType.StartArray || tokenType == JsonTokenType.StartObject) ==
                     (length == DbRow.UnknownSize));
 
-#if DEBUG
-                Debug.Assert(!_isLocked, "Appending to a locked database");
```

Comment:
Why did we remove this `Debug.Assert`? Add it back.

Reply:
The Assert was moved into the `Enlarge()` that this method  (`Append`) calls.

The serializer fast-path calls into this method to append, but will never pass the `if` check just after this and call into `Enlarge`.

Reply:
Any harm it keeping it here too? If not, we should keep it to be clear about the expected invariant, in case the code evolves.

Reply:
The first line in `Enlarge()` does the Assert, and `Enlarge()` is only called from this one location. I don't see value in duplicating the Assert or moving it up.

However, perhaps removing `Enlarge()` is better.

Reply:
I see. Removing the extra debug.assert makes sense. Thanks for the explanation.

Reply:
I don't think it is an inlining issue since no automatic inlining is done when there is any condition AFAIK. However, I added back the Enlarge anyway.